bson/src/main/org/bson/codecs/jsr310/DateTimeBasedCodec.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bson.codecs.jsr310;
+
+import org.bson.BsonReader;
+import org.bson.BsonType;
+import org.bson.codecs.Codec;
+import org.bson.codecs.configuration.CodecConfigurationException;
+
+import static java.lang.String.format;
+
+abstract class DateTimeBasedCodec<T> implements Codec<T> {
+
+    long validateAndReadDateTime(final BsonReader reader) {
+        BsonType currentType = reader.getCurrentBsonType();
+        if (currentType == null) {
```

Comment:
I thought that a Codec could assume that the BsonReader is positioned correctly at the type. Under what circumstances would this conditional be entered?

Reply:
Copy Pasta - updated.